src/test_fixture/world.rs

```diff
@@ -125,4 +161,70 @@ where
             join_all(zip(contexts, input_shares).map(|(ctx, shares)| helper_fn(ctx, shares))).await;
         <[_; 3]>::try_from(output).unwrap()
     }
+
+    async fn malicious<'a, F, O, M, H, R, P>(&'a self, input: I, mut helper_fn: H) -> [O; 3]
+    where
+        A: IntoMalicious<F, M>,
+        F: Field,
+        O: Send + Debug,
+        M: Send,
+        H: FnMut(MaliciousContext<'a, F>, M) -> R + Send,
+        R: Future<Output = P> + Send,
+        P: DowngradeMalicious<Target = O> + Send + Debug,
+        Standard: Distribution<F>,
+    {
+        // The following is what this *should* look like,
+        // but so far the spelling necessary to convince the borrow checker
+        // to accept this has not been found.
+        //
+        // Current theory is that this might allow `helper_fn` to be run
+        // on multiple different threads concurrently, which would be bad.
+        // The long form below ensures that it is only run on one thread,
+        // even if it might move (with `Send`) a few times before it runs.
+
+        #[cfg(exemplary_code)]
+        {
+            self.semi_honest(input, |ctx, share| async {
+                let v = MaliciousValidator::new(ctx);
+                let m_share = share.upgrade(v.context()).await;
```

Comment:
I find it very strange to invoke `upgrade` on the share - rather than the context. Why not just call:

```
let m_share = v.context().upgrade(RecordId::from(i), share).await;
```

Reply:
Yeah, this is a consequence of the `IntoMalicious` trait having to be implemented on the share.  Maybe we can work something else out, but I can't see how to do this easily.  One thing that makes this harder is that the trait is a test-only thing right now, because it needs to manage record IDs.